src/ui/src/builder/BuilderJournal.vue

```diff
@@ -1,11 +1,205 @@
 <template>
-	<div class="BuilderJournal"></div>
+	<div class="BuilderJournal">
+		<BuilderJournalHeader
+			@refresh="loadEntries"
+			@clear="deleteEntries"
+			@download="downloadAsJson"
+			@update:search="(val) => (searchText = val)"
+			@update:statuses="(val) => (selectedStatuses = val)"
+			@update:triggers="(val) => (selectedTriggers = val)"
+		></BuilderJournalHeader>
+		<div class="BuilderJournal__entries">
+			<BuilderJournalEntry
+				v-for="(entry, key) of sortedEntries"
+				:key
+				:journal-entry="entry"
+			></BuilderJournalEntry>
+		</div>
+	</div>
 </template>
 
+<script setup lang="ts">
+import { computed, ComputedRef, inject, onMounted, Ref, ref } from "vue";
+import BuilderJournalHeader from "./journal/BuilderJournalHeader.vue";
+import BuilderJournalEntry from "./journal/BuilderJournalEntry.vue";
+import { convertAbsolutePathtoFullURL } from "@/utils/url";
+import { useToasts } from "./useToast";
+import { Component, WriterComponentDefinition } from "@/writerTypes";
+import injectionKeys from "@/injectionKeys";
+
+const { pushToast } = useToasts();
+const wf = inject(injectionKeys.core);
+
+type ComponentInfo = {
+	type: "blueprint" | "block";
+	id: string;
+};
+type TriggerInfo = {
+	type: "On demand" | "UI" | "API" | "Cron";
+	event: string;
+	component: ComponentInfo;
+};
+
+type BlockOutput = {
+	result: any;
+	outcome: string;
+};
+
+export type RawJournalEntry = {
+	trigger: TriggerInfo;
+	timestamp: string;
+	blockOutputs: Record<string, BlockOutput>;
+	instanceType: "editor" | "agent";
+	result: "success" | "error" | "stopped";
+};
+
+export type JournalEntry = RawJournalEntry & {
+	title: string;
+	component: Component;
+	componentDefinition: WriterComponentDefinition;
+};
+
+const rawEntries: Ref<Record<string, RawJournalEntry>> = ref({});
+
+const searchText = ref("");
+const selectedStatuses = ref<string[]>([]);
+const selectedTriggers = ref<string[]>([]);
+
+const entries: ComputedRef<Record<string, JournalEntry>> = computed(() => {
+	return Object.fromEntries(
+		Object.entries(rawEntries.value).map(([key, entry]) => {
+			const component = wf.getComponentById(entry.trigger.component.id);
+			const componentDefinition = wf.getComponentDefinition(
+				component.type,
+			);
+
+			const title =
+				component.type === "blueprints_blueprint"
+					? component.content.key
+					: component.content.alias || componentDefinition.name;
+
+			return [key, { ...entry, title, component, componentDefinition }];
+		}),
+	);
+});
+
+const filteredEntries: ComputedRef<Record<string, JournalEntry>> = computed(
+	() => {
+		return Object.fromEntries(
+			Object.entries(entries.value).filter(([_, entry]) => {
+				const searchMatch =
+					searchText.value === "" ||
+					entry.title
+						.toLowerCase()
+						.includes(searchText.value.toLowerCase());
```

Comment:
It's better to keep `searchText.value.toLowerCase()` outside of the loop

```suggestion
    const searchTextLower = searchText.value.toLowerCase();
		return Object.fromEntries(
			Object.entries(entries.value).filter(([_, entry]) => {
				const searchMatch =
					searchText.value === "" ||
					entry.title
						.toLowerCase()
						.includes(searchTextLower);
```